lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -1551,33 +1614,66 @@ binaryCalculationsSpec' era = describe ("calculateBinary - "+||era||+"") $ do
+                        \e82201d7072e6fbfc0a1884fb54636554d14945b799125cf7ce38d\
+                        \477f515840320ed7d1513b0f1b61381f7942a07b627b246c85a13b\
+                        \2623e4868ea82488c778a7760124f3a17f924c08d425c0717df6cd\
+                        \898eb4ab8439a16e08befdc415120e582001010101010101010101\
+                        \0101010101010101010101010101010101010101010144a1024100\
+                        \845820010000000000000000000000000000000000000000000000\
+                        \000000000000000058401a8667d2d0af4e24d4d385443002f1e903\
+                        \6063bdb7c62d45447a2e176ded81a11683bd944c6d7db6e5fd8868\
+                        \40025f63192a382e526f4150e2b336ee9ed8080858200000000000\
+                        \000000000000000000000000000000000000000000000000000000\
+                        \44a1024100f5f6"
+                    _ ->
+                        "83a400828258200000000000000000000000000000000000000000\
+                        \000000000000000000000000008258200000000000000000000000\
+                        \000000000000000000000000000000000000000000010183825839\
+                        \010202020202020202020202020202020202020202020202020202\
+                        \020202020202020202020202020202020202020202020202020202\
+                        \0202021a005b8d8082583901030303030303030303030303030303\
+                        \030303030303030303030303030303030303030303030303030303\
+                        \03030303030303030303030303031a005b8d808258390104040404\
+                        \040404040404040404040404040404040404040404040404040404\
+                        \040404040404040404040404040404040404040404040404041a00\
+                        \7801e0021a0002102003191e46a10282845820130ae82201d7072e\
+                        \6fbfc0a1884fb54636554d14945b799125cf7ce38d477f51584058\
+                        \35ff78c6fc5e4466a179ca659fa85c99b8a3fba083f3f3f42ba360\
+                        \d479c64ef169914b52ade49b19a7208fd63a6e67a19c406b482660\
+                        \8fdc5307025506c307582001010101010101010101010101010101\
+                        \0101010101010101010101010101010144a1024100845820010000\
+                        \000000000000000000000000000000000000000000000000000000\
+                        \00005840e8e769ecd0f3c538f0a5a574a1c881775f086d6f4c845b\
+                        \81be9b78955728bffa7efa54297c6a5d73337bd6280205b1759c13\
+                        \f79d4c93f29871fc51b78aeba80e58200000000000000000000000\
+                        \00000000000000000000000000000000000000000044a1024100f6"
+
+            calculateBinary net utxo outs chgs pairs `shouldBe` binary
 
   where
     slotNo = SlotNo 7750
     md = Nothing
     calculateBinary net utxo outs chgs pairs =
-        hex (Cardano.serialiseToCBOR ledgerTx)
+        case era of
+            ShelleyBasedEraAlonzo ->
+                slimCBOR eraSerializedCBOR
```

Comment:
Is it not saner to also add Alonzo-specific expected values in your case expressions above, like you already have for Babbage?

Reply:
In fact, the Alonzo and Babbage binaries are the same, no? I haven't run and tested, but it looks like https://github.com/input-output-hk/cardano-ledger/pull/2863 should have affected Alonzo as well.

Then having the same expected value for Alonzo and Babbage sounds much easier, and you could still keep your comment explaining the difference.

Reply:
thanks Johannes for your look. I tend to agree and now we have binary rep for Babbage, Alonzo and the rest. There are differences between all of them. I added the comment illustrating the point in the relevant test:
```
1421 -- Up till Mary era we have the following structure of transaction                                                                                                                      
1422 --   transaction =                                                                                                                                                                      
1423 -- [ transaction_body                                                                                                                                                                   
1424 -- , transaction_witness_set                                                                                                                                                            
1425 -- , auxiliary_data / null                                                                                                                                                              
1426 -- ]                                                                                                                                                                                    
1427 -- So we begin with 3-element array binary prefix, that is encoded as '83'                                                                                                              
1428 -- From Alonzo on tx was enriched for isValid field making it                                                                                                                           
1429 -- 4-element array that is encoded as '84'.                                                                                                                                             
1430 --   transaction =                                                                                                                                                                      
1431 -- [ transaction_body                                                                                                                                                                   
1432 -- , transaction_witness_set                                                                                                                                                            
1433 -- , bool                                                                                                                                                                               
1434 -- , auxiliary_data / null                                                                                                                                                              
1435 -- ]                                                                                                                                                                                    
1436 -- Alonzo transaction stil has the same binary representation (array)                                                                                                                   
1437 -- for transaction outputs like in the previous eras.                                                                                                                                   
1438 -- Babbage era changes this representation, and introduces map binary                                                                                                                   
1439 -- representation for transaction outputs as the contept of them is                                                                                                                     
1440 -- extended in this era.                                    
```

Reply:
Ah ok so they are different after all 👌